### ISSUE TITLE
Missing measurements in metrics of nomad server when scraping with prometheus

### ISSUE THREAD

**Issue by parberge** [hashicorp/nomad#5903]
### Nomad version
Nomad v0.9.3

### Operating system and Environment details
Ubuntu 16.04

### Issue
The metric "nomad_nomad_job_summary_queued" sometime is not present during scraping.
![image](https://user-images.githubusercontent.com/5441497/60337953-bd3ba600-99a4-11e9-82cf-f2e99da34569.png)
As seen in the screenshot from prometheus quite often it doesn't show any measurement.  It should always produce a value.

### Reproduction steps
Scrape metrics endpoint of nomad servers with prometheus using `format=prometheus`

### Nomad Server logs (if appropriate)
This is logged on the nomad server that probably is related to the problem
```
* collected metric nomad_nomad_job_summary_queued label:<name:"job" value:"xx:xx/periodic-1561711500" > label:<name:"parent_id" value:"xx:xx" > label:<name:"periodic_id" value:"1561711500" > label:<name:"task_group" value:"xx-xx" > gauge:<value:0 >  has label dimensions inconsistent with previously collected metrics in the same metric family
```

We have these `telemetry` settings on each nomad server
```
telemetry {
  publish_allocation_metrics = true
  publish_node_metrics       = true
  prometheus_metrics         = true
  disable_hostname           = true
}
```

**Comment by parberge** [comment]
Might be related to https://github.com/hashicorp/nomad/issues/5345 ?

**Comment by parberge** [comment]
We have 3 nomad servers that we scrape. I just checked, and it seems that only the leader has these measurements. Is that expected behaviour?

**Comment by parberge** [comment]
Just checked the documentation and you have specified that only the leader will omit this.
I will just close this then. Sorry for the fuss 😏

**Comment by parberge** [comment]
Just realised that it shouldn't matter even if we scrape the non-leaders, since we always scrape the leader.

This should still always produce values.

**Comment by parberge** [comment]
traversed the time series, and saw this:
Sometimes it shows only the periodic jobs in the output, but usually it shows the non-periodic jobs in the output. Never both at the same time. I guess it can be the reason for "has label dimensions inconsistent" errors and maybe also the reason why prometheus doesn't pickup the value?

**Comment by parberge** [comment]
This seems to have been solved in nomad `0.9.4`
